The `WebTestClient` throws exceptions with little info

The `WebTestClient` throws a generic instruction when the strategy fails with no indication on what the error actually is. It would be helpful to minimally log the response or append the response text to the failure message.

The current message is:

```
[ERROR] ee.jakarta.tck.pages.spec.tagext.resource.httplistener.URLClientIT.testResourceSL -- Time elapsed: 0.091 s <<< ERROR!
java.lang.Exception: [BaseUrlClient] testResourceSL failed!  Check output for cause of failure.
	at ee.jakarta.tck.pages.common.client.BaseUrlClient.invoke(BaseUrlClient.java:464)
	at ee.jakarta.tck.pages.spec.tagext.resource.httplistener.URLClientIT.testResourceSL(URLClientIT.java:121)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at org.jboss.arquillian.junit5.ArquillianExtension$1.invoke(ArquillianExtension.java:198)
	at org.jboss.arquillian.container.test.impl.execution.LocalTestExecuter.execute(LocalTestExecuter.java:57)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:85)
	at org.jboss.arquillian.core.impl.EventContextImpl.invokeObservers(EventContextImpl.java:102)
	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:89)
	at org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:133)
	at org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:105)
	at org.jboss.arquillian.core.impl.EventImpl.fire(EventImpl.java:61)
	at org.jboss.arquillian.container.test.impl.execution.ClientTestExecuter.execute(ClientTestExecuter.java:49)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:85)
	at org.jboss.arquillian.core.impl.EventContextImpl.invokeObservers(EventContextImpl.java:102)
	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:89)
	at org.jboss.arquillian.container.test.impl.client.ContainerEventController.createContext(ContainerEventController.java:127)
	at org.jboss.arquillian.container.test.impl.client.ContainerEventController.createTestContext(ContainerEventController.java:117)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:85)
	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:94)
	at org.jboss.arquillian.test.impl.TestContextHandler.createTestContext(TestContextHandler.java:115)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:85)
	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:94)
	at org.jboss.arquillian.test.impl.TestContextHandler.createClassContext(TestContextHandler.java:82)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:85)
	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:94)
	at org.jboss.arquillian.test.impl.TestContextHandler.createSuiteContext(TestContextHandler.java:68)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:85)
	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:94)
	at org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:133)
	at org.jboss.arquillian.test.impl.EventTestRunnerAdaptor.test(EventTestRunnerAdaptor.java:138)
	at org.jboss.arquillian.junit5.ArquillianExtension.interceptInvocation(ArquillianExtension.java:178)
	at org.jboss.arquillian.junit5.ArquillianExtension.interceptTestMethod(ArquillianExtension.java:118)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
Caused by: com.sun.ts.tests.common.webclient.TestFailureException: Test FAILED!
	at com.sun.ts.tests.common.webclient.WebTestCase.execute(WebTestCase.java:173)
	at ee.jakarta.tck.pages.common.client.BaseUrlClient.invoke(BaseUrlClient.java:455)
	... 39 more
```

There is no valuable output in the logs, even with TRACE enabled.
```
************************************************************
* props file set to "/tmp/jperkins-cts-props.txt"
************************************************************
[client] 07:37:55,859 FINE  [ee.jakarta.tck.pages.common.client.BaseUrlClient] (main) [BaseUrlClient] EXECUTING
[client] 07:37:55,886 WARN  [org.apache.commons.httpclient.HttpConnection] (main) HttpConnectionManager is null.  Connection cannot be released.
[client] 07:37:55,891 INFO  [ee.jakarta.tck.pages.spec.tagext.resource.httplistener.URLClientIT] (main) FINISHED TEST : testResourceSL()
[client] 07:37:55,905 INFO  [ee.jakarta.tck.pages.common.client.ServletAbstractUrlClient] (main) setup method ServletAbstractUrlClient
[client] 07:37:55,905 INFO  [ee.jakarta.tck.pages.common.client.ServletAbstractUrlClient] (main) [ServletAbstractUrlClient] Test setup OK
[client] 07:37:55,906 INFO  [ee.jakarta.tck.pages.spec.tagext.resource.httplistener.URLClientIT] (main) STARTING TEST : testResourceSAL()
[client] 07:37:55,908 FINE  [ee.jakarta.tck.pages.common.client.BaseUrlClient] (main) [BaseUrlClient] EXECUTING
[client] 07:37:55,910 WARN  [org.apache.commons.httpclient.HttpConnection] (main) HttpConnectionManager is null.  Connection cannot be released.
[client] 07:37:55,911 INFO  [ee.jakarta.tck.pages.spec.tagext.resource.httplistener.URLClientIT] (main) FINISHED TEST : testResourceSAL()
```

This is what currently happens:

```java
if (!_strategy.validate(this)) {
    throw new TestFailureException("Test FAILED!");
}
``` 

Some tests might expect that explicit message at first glance which is why we might need to simply log the response again or log the response if DEBUG logging is enabled.

--- a/tools/common/src/main/java/com/sun/ts/tests/common/webclient/WebTestCase.java
+++ b/tools/common/src/main/java/com/sun/ts/tests/common/webclient/WebTestCase.java
@@ -26,6 +26,7 @@ import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 import java.util.StringTokenizer;
+import java.util.logging.Level;
 import java.util.logging.Logger;
 
 import com.sun.ts.lib.util.TestUtil;
@@ -165,11 +166,12 @@ public class WebTestCase implements TestCase {
         } catch (Throwable t) {
             String message = t.getMessage();
             throw new TestFailureException(
-                    "[FATAL] Unexpected failure during " + "test execution." + (message == null ? t.toString() : message), t);
+                    "[FATAL] Unexpected failure during test execution." + (message == null ? t.toString() : message), t);
         }
 
         // Validate this test case instance
         if (!_strategy.validate(this)) {
+            LOGGER.log(Level.SEVERE, String.format("[FAILED] Request: %s%nResponse: %s", _request, _response));
             throw new TestFailureException("Test FAILED!");
         }
 

